plenum/server/primary_selector.py

```diff
@@ -54,14 +54,56 @@ def next_primary_node_name(self, instance_id, nodeReg=None):
 
         return name
 
-    def next_primary_replica_name(self, instance_id, nodeReg=None):
+    def next_primary_node_name_for_master(self, nodeReg=None):
+        if nodeReg is None:
+            nodeReg = self.node.nodeReg
+        rank = self._get_primary_id(self.viewNo, 0, len(nodeReg))
+        name = self.node.get_name_by_rank(rank, nodeReg=nodeReg)
+
+        logger.trace("{} selected {} as next primary node for master instance, "
+                     "viewNo {} with rank {}, nodeReg {}".format(
+                         self, name, self.viewNo, rank, nodeReg))
+        assert name, "{} failed to get next primary node name for master instance".format(self)
+
+        return name
+
+    def next_primary_node_name_for_backup(self, instance_id, nodeReg=None):
+        if nodeReg is None:
+            nodeReg = self.node.nodeReg
+        rank = self._get_primary_id(self.viewNo, instance_id, len(nodeReg))
+        name = self.node.get_name_by_rank(rank, nodeReg=nodeReg)
+
+        logger.trace("{} selected {} as next primary node for instId {}, "
+                     "viewNo {} with rank {}, nodeReg {}".format(
+                         self, name, instance_id, self.viewNo, rank, nodeReg))
+        assert name, "{} failed to get next primary node name".format(self)
+
+        return name
+
+    def next_primary_replica_name_for_master(self, nodeReg=None):
         """
         Returns name of the next node which is supposed to be a new Primary
```

Comment:
Please mention that we return two values, not just name.

Reply:
Done.